packages/server/src/manifest-builder.ts

```diff
@@ -63,26 +64,47 @@ function* processManifest(options: ManifestBuilderOptions): Operation {
   return distPath;
 }
 
+function logBuildError(error: BundlerError) {
+  console.error("[manifest builder] build error:", error.message);
+  console.error("[manifest builder] build error frame:\n", error.frame);
+}
+
+function* waitForSuccessfulBuild(bundlerEvents: ChainableSubscription<BundlerMessage, undefined>, delegate: Mailbox): Operation {
+  let message = yield bundlerEvents.first();
+
+  if (message.type === "error") {
+    logBuildError(message.error);
+    delegate.send({ event: "error" });
+    yield waitForSuccessfulBuild(bundlerEvents, delegate);
+  }
+}
+
 export function* createManifestBuilder(options: ManifestBuilderOptions): Operation {
-  let parcel: ParcelProcess = yield ParcelProcess.create(
-    [options.srcPath],
-    {
-      outDir: options.buildDir,
-      global: bigtestGlobals.manifestProperty,
-      outFile: "manifest.js"
-    }
+  let bundler: Bundler = yield Bundler.create(
+    [{
+      entry: options.srcPath,
+      globalName: bigtestGlobals.manifestProperty,
+      outFile: path.join(options.buildDir, "manifest.js")
+    }]
   );
+  let bundlerEvents: ChainableSubscription<BundlerMessage, undefined> = yield subscribe(bundler);
```

Comment:
One (fixable) problem with using `Subscribable.from()` is the `Chain` interface is not public, making type annotations with it impossible. This is not a great reason, but it was my deciding factor in this case. I do feel that `subscribe()` may be a better fit though as we would otherwise be recursively creating new subscriptions in `waitForSuccessfulBuild()` if I understand correctly.

As an aside, the overlap between `Subscribable` and `Chain` and `ChainableSubscription` is confusing to me. Not sure if there's room to make the differences more clear and understandable or otherwise consolidate into fewer interfaces.

Reply:
I initially wanted to remove `Subscribable` entirely, but we left it in there, because we didn't know which type of chaining would turn out to be more useful, so we decided on keeping both. That being said, we could certainly improve on the naming of things to make it easier to understand.

Reply:
@pittst3r seems like the right choice for a recursive method, although this is a very subtle distinction to ask users to 

I wonder if we can make some shared interface that has all the chaining, and then have the low-level thing that just has `next()` on it a `SubscriptionIterator`. We might be able to use that chaining interface regardless of whether we're chaining the subscription or the operation.